pkg/azure/provider/provider_test.go

```diff
@@ -0,0 +1,1180 @@
+			}
+			clusterState.WithDefaultVMImageSpec().WithAgreementTerms(true).WithSubnet(providerSpec.ResourceGroup, fakes.CreateSubnetName(testShootNs), testShootNs)
+			// create fake factory
+			fakeFactory := createFakeFactoryForCreateMachineWithAPIBehaviorSpecs(g, providerSpec.ResourceGroup, clusterState, entry.vmAccessAPIBehavior, nil, entry.nicAccessAPIBehavior, nil, nil)
+
+			// Create machine and machine class to be used to create DeleteMachineRequest
+			machineClass, err := fakes.CreateMachineClass(providerSpec, to.Ptr(testResourceGroupName))
+			g.Expect(err).To(BeNil())
+			machine := &v1alpha1.Machine{
+				ObjectMeta: fakes.NewMachineObjectMeta(testShootNs, vmName),
+			}
+			// Test
+			//----------------------------------------------------------------------------
+			testDriver := NewDefaultDriver(fakeFactory)
+			_, err = testDriver.CreateMachine(ctx, &driver.CreateMachineRequest{
+				Machine:      machine,
+				MachineClass: machineClass,
+				Secret:       fakes.CreateProviderSecret(),
+			})
+			if entry.checkClusterStateFn != nil {
+				entry.checkClusterStateFn(g, ctx, clusterState, vmName)
+			}
+			if entry.checkErrorFn != nil {
+				entry.checkErrorFn(g, clusterState, err)
+			}
+		})
+	}
+}
+
+func TestSuccessfulCreationOfMachine(t *testing.T) {
+	g := NewWithT(t)
+	providerSpecBuilder := testhelp.NewProviderSpecBuilder(testResourceGroupName, testShootNs, testWorkerPool0Name).
+		WithDefaultValues().
+		WithDataDisks(testDataDiskName, 2)
+	providerSpec := providerSpecBuilder.Build()
+
+	// initialize cluster state
+	//----------------------------------------------------------------------------
+	// create cluster state
+	clusterState := fakes.NewClusterState(providerSpec)
+	clusterState.WithDefaultVMImageSpec().WithAgreementTerms(true).WithSubnet(providerSpec.ResourceGroup, fakes.CreateSubnetName(testShootNs), testShootNs)
+	// create fake factory
+	fakeFactory := createFakeFactoryForCreateMachineWithAPIBehaviorSpecs(g, providerSpec.ResourceGroup, clusterState, nil, nil, nil, nil, nil)
+	// Create machine and machine class to be used to create DeleteMachineRequest
+	machineClass, err := fakes.CreateMachineClass(providerSpec, to.Ptr(testResourceGroupName))
+	const vmName = "vm-0"
+	g.Expect(err).To(BeNil())
+	ctx := context.Background()
+	machine := &v1alpha1.Machine{
+		ObjectMeta: fakes.NewMachineObjectMeta(testShootNs, vmName),
+	}
+	dataDiskNames := testhelp.CreateDataDiskNames(vmName, providerSpec)
+
+	// Test
+	//----------------------------------------------------------------------------
+	testDriver := NewDefaultDriver(fakeFactory)
+	resp, err := testDriver.CreateMachine(ctx, &driver.CreateMachineRequest{
+		Machine:      machine,
+		MachineClass: machineClass,
+		Secret:       fakes.CreateProviderSecret(),
+	})
+	g.Expect(err).To(BeNil())
+	checkClusterStateAndGetMachineResources(g, ctx, *fakeFactory, vmName, true, true, true, dataDiskNames, true, false)
```

Comment:
```suggestion
	checkClusterStateAndGetMachineResources(g, ctx, *fakeFactory, vmName, true, true, true, dataDiskNames, true, true)
```
Since the created disks and Nics would have VM ref

Reply:
Making this true, resulted in a test failure. This pointed to a bug in clusterstate.go/CreateVM function. Corrected that now.